build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle.kts

```diff
@@ -85,4 +88,18 @@ class Capture(val logger: Logger) {
                 }
         }
     }
+
+    private val filterIncompatibleKotestVersions = Predicate<File> { jar ->
+        if (jar.name.contains("kotest-runner")) {
+            val kotestVersionString = jar.name.split("-")[jar.name.split("-").lastIndex].replace(".jar", "")
+            val kotestVersion = VersionNumber.parse(kotestVersionString)
+            if (VersionNumber.UNKNOWN == kotestVersion) {
+                logger.warn("Unable to parse kotest version from file name ${jar.name}")
```

Comment:
im wondering if we should put this as `error` level or just directly throw an exception so we can fix the error. we could also add a custom value here as well to prevent just thinking it was incompatible.

Reply:
Good point, I think an exception or custom value (since we are looking at custom values, and maybe not logs, when determining compatibility) would be best so that there is less of a chance the issue is missed.

Reply:
Pushed a commit adding a custom value and logging the error, required a bit of rework to make the `BuildScanExtension` and `Test` task available to the predicate - lmk if you think this is overkill

Reply:
Here's an example of how this would look https://ge.solutions-team.gradle.com/s/2ukw7odlyyge4/custom-values#L2 (i set the version to unknown to force the custom value to be created, normally this 5.7.2 version would be compatible)

Reply:
Thanks for the example, I think that will help in debugging!